lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2085,26 +2085,45 @@ Instruction *SPIRVToLLVM::transEnqueueKernelBI(SPIRVInstruction *BI,
   // Find or create enqueue kernel BI declaration
   auto Ops = BI->getOperands();
   bool HasVaargs = Ops.size() > 10;
+  bool HasEvents = true;
+  SPIRVValue *NumEvents = Ops[3];
+  if (NumEvents->getOpCode() == OpConstant) {
+    SPIRVConstant *NE = static_cast<SPIRVConstant *>(NumEvents);
+    HasEvents = NE->getZExtIntValue() != 0;
+  }
+
+  std::string FName = "";
+  if (!HasVaargs && !HasEvents)
+    FName = "__enqueue_kernel_basic";
+  else if (!HasVaargs && HasEvents)
+    FName = "__enqueue_kernel_basic_events";
+  else if (HasVaargs && !HasEvents)
+    FName = "__enqueue_kernel_varargs";
+  else
+    FName = "__enqueue_kernel_events_varargs";
 
-  std::string FName = HasVaargs ? "__enqueue_kernel_events_varargs"
-                                : "__enqueue_kernel_basic_events";
   Function *F = M->getFunction(FName);
   if (!F) {
-    Type *EventTy = PointerType::get(
-        getOrCreateOpaquePtrType(M, SPIR_TYPE_NAME_CLK_EVENT_T,
-                                 getOCLOpaqueTypeAddrSpace(OpTypeDeviceEvent)),
-        SPIRAS_Generic);
-
-    SmallVector<Type *, 8> Tys = {
+    SmallVector<Type *, 3> Tys = {
```

Comment:
I don't think you should change the size to 3, as this SmallVector will always contain more than 3 elements (which will cause a heap allocation).